### ISSUE TITLE
Add support for SpiderMonkey 78

### ISSUE THREAD

**Issue by wohali** [apache/couchdb#3476]
Necessary for debian bullseye, which is now in hard freeze:

https://packages.debian.org/search?keywords=mozjs&searchon=names&suite=testing&section=all

This will incidentally allow compiling against Ubuntu `groovy (20.10)` and `hirsuite`, though these are not officially supported releases, Fedora 34, and Alpine Linux 3.13. (CentOS 8 "Stream" is still on mozjs 52/60.)

Hopefully this is "most of, but not all" of the changes @janl landed in #3434 plus a couple of minor `rebar.config.script` tweaks.

**Comment by dch** [comment]
yes please, as FreeBSD is dropping support for any python2 packages in late May timeframe (before 2021Q3 is branched), https://www.freshports.org/lang/spidermonkey78 would be the preferred SpideyCouch target, I will ask about SM86 support which FreeBSD doesn't yet provide.

```
$ pwd
/usr/ports/lang/spidermonkey78
$ make fetch-url-list

https://download.cdn.mozilla.net/pub/firefox/releases/78.7.0esr/source/firefox-78.7.0esr.source.tar.xz
https://archive.mozilla.org/pub/firefox/releases/78.7.0esr/source/firefox-78.7.0esr.source.tar.xz
http://distcache.FreeBSD.org/ports-distfiles/firefox-78.7.0esr.source.tar.xz
```

**Comment by wohali** [comment]
Closed by #3500.
